perfkitbenchmarker/linux_packages/memcached_server.py

```diff
@@ -0,0 +1,121 @@
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+
+"""Module containing memcached server installation and cleanup functions."""
+
+import logging
+
+from perfkitbenchmarker import errors
+from perfkitbenchmarker import flags
+from perfkitbenchmarker import vm_util
+from perfkitbenchmarker.linux_packages import INSTALL_DIR
+
+FLAGS = flags.FLAGS
+
+LATEST_URL = 'http://memcached.org/latest'
+MEMCACHED_DIR_NAME = 'memcached'
+MEMCACHED_DIR = '%s/%s' % (INSTALL_DIR, MEMCACHED_DIR_NAME)
+
+MEMCACHED_PORT = 11211
+
+flags.DEFINE_integer('memcached_size', 64,
+                     'Size of memcached cache in megabytes.')
+
+
+def _Install(vm):
+  """Installs the memcached server on the VM."""
+  vm.Install('build_tools')
+  vm.Install('event')
+  vm.RemoteCommand('cd {0} && wget {1} -O memcached.tar.gz'.format(
+                   INSTALL_DIR, LATEST_URL))
+  out, _ = vm.RemoteCommand('cd %s && tar -xzvf memcached.tar.gz' % INSTALL_DIR)
+  # The directory name should be the first line of stdout
+  memcached_dir = out.split('\n', 1)[0]
+  # Rename the directory to a standard name
+  vm.RemoteCommand('cd {0} && mv {1} {2}'.format(
+                   INSTALL_DIR, memcached_dir, MEMCACHED_DIR_NAME))
+  # Make memcached
+  vm.RemoteCommand('cd {0} && ./configure && make'.format(MEMCACHED_DIR))
+
+
+def YumInstall(vm):
+  """Installs the memtier package on the VM."""
+  _Install(vm)
+
+
+def AptInstall(vm):
+  """Installs the memtier package on the VM."""
+  _Install(vm)
+
+
+@vm_util.Retry(poll_interval=5, timeout=300,
+               retryable_exceptions=(errors.Resource.RetryableCreationError))
+def _WaitForServerUp(server):
+  """Block until the memcached server is up and responsive.
+
+  Will timeout after 5 minutes, and raise an exception. Before the timeout
+  expires any exceptions are caught and the status check is retried.
+
+  We check the status of the server by issuing a 'stats' command. This should
+  return many lines of form 'STAT <name> <value>\\r\\n' if the server is up and
+  running.
+
+  Args:
+    server: VirtualMachine memcached has been installed on.
+
+  Raises:
+    errors.Resource.RetryableCreationError when response is not as expected or
+      if there is an error connecting to the port or otherwise running the
+      remote check command.
+  """
+  address = server.internal_ip
+  port = MEMCACHED_PORT
+
+  logging.info("Trying to connect to memcached at %s:%s" % (address, port))
```

Comment:
for logging, you can do logging.info("Trying to connect to memcached at %s:%s" , address, port)